### PR TITLE
set current position to actual device position

### LINKED ISSUE #14: Device state is not checked
When HomeKit performs a state update, the device is not checked for the state and only relies on the initial default "OFF" state. This results in all devices being "OFF" if Homebridge is restarted.

The `targetState` variable is declared but never used:
https://github.com/SwitchBot/homebridge-switchbot-ble/blob/ec96d90aac9da6e8d200c25045f1bac9f7fbe809/src/bot-accessory.ts#L40

Would a pull request be accepted that checks the device (at least the initial state) by checking the [advertisement](https://github.com/futomi/node-switchbot#monitoring-advertising-packets) data?

### PR DESCRIPTION
**Pull request by marzocchi**
This PR resolves #14 by setting `currentPosition` (and `targetPosition`) to the position reported by the device advertisements using timed scans (every `60s` when idle and every `0.5s` when opening/closing via HomeKit).

### CODE REVIEW

File: src/curtain-accessory.ts [hunk 1]
```diff
@@ -10,9 +10,162 @@ import {
...
+        .catch((err: any) => {
+          reject(err);
+        });
+    });
+  }
+
+  private applyPosition(ad: any) {
+    let pos = ad.serviceData.position;
+
+    if (pos >= this.considerClosedAtPosition) {
```
Comment (marzocchi) [inline src/curtain-accessory.ts:147]: this replicates the behaviour I see also with the official SwitchBot iOS app: the curtain is reported as fully closed even though the position reported by the device is 98% (even after repeated calibrations). what's the best channel to report issues with the device?

File: src/curtain-accessory.ts [hunk 2]
```diff
@@ -10,9 +10,162 @@ import {
...
+    this.scanDuration = scanDuration;
+    this.slowScanInterval = scanInterval;
+    this.callback = callback;
+
+    this.startFastScan();
+  }
+
+  /**
+   * Ask the device to start moving to the given position (which must be a device value and not an HomeKit value).
+   * Returns a Promise that's resolved as soon as the command was sent to the device.
```
Comment (marzocchi) [inline src/curtain-accessory.ts:52]: I did not include it in this PR to limit the diff size, but the next step would be to make this Promise resolve only when the position reported by the curtain is equal to `pos` and do away with `Curtain.moveTimer`. @donavanbecker, any thoughts?

Comment (donavanbecker) [reply src/curtain-accessory.ts:52]: @marzocchi submit the pr and I will merge it into beta branch to publish beta version to testing.

File: src/curtain-accessory.ts (hunk 1 above)
Comment (donavanbecker) [reply src/curtain-accessory.ts:147]: SwitchBot App.

**Review by donavanbecker — APPROVED** [review]

File: src/curtain-accessory.ts (hunk 2 above)
Comment (marzocchi) [reply src/curtain-accessory.ts:52]: will do, thanks!

**Pull request merged**